ratatui-widgets/src/table.rs

```diff
@@ -910,6 +910,11 @@ impl Table<'_> {
     fn visible_rows(&self, state: &TableState, area: Rect) -> (usize, usize) {
         let last_row = self.rows.len().saturating_sub(1);
         let mut start = state.offset.min(last_row);
+
+        if let Some(selected) = state.selected {
+            start = start.min(selected);
+        }
+
```

Comment:
```suggestion
        let mut start = state
            .selected
            .map_or(state.offset, |i| state.offset.min(i))
            .min(last_row);
```

Reply:
I think this is way harder to read, even when `i` is replaced with `selected`